Chrysalis-Papilio/Soen-490-Papilio-Backend
# Chat routes implementation

# General info

**Issue number**: PAPILIO-30

**Task description**: 
 - Added chat routes to create a new channel, add users to existing channels, leave existing channels, remove a channel
 - Users have the ability to join/leave an activity including the channel

# Testing
To test this feature you have to (I have tested all routes so looking at the code should be enough):

1. Add these two environment variables in the .env file:

        STREAM_CHAT_API_KEY=
        STREAM_CHAT_API_SECRET=
2. Create a new activity and verify that a new chat has been created within the StreamChat dashboard with id being the same as the newly created activity id
3. It is not necessary to test all routes related to StreamChat since I have already tested them on android and postman and I don't want to polute StreamChat with useless users and channels.

# Notes
- Code was tested with Postman and the android mobile app
